core/sql/generator/GenRelScan.cpp

```diff
@@ -1119,7 +1124,8 @@ short FileScan::codeGenForHive(Generator * generator)
   if (expirationTimestamp > 0)
     expirationTimestamp += 1000000 *
       (Int64) CmpCommon::getDefaultLong(HIVE_METADATA_REFRESH_INTERVAL);
-  generator->setPlanExpirationTimestamp(expirationTimestamp);
+  if (!getCommonSubExpr())
+    generator->setPlanExpirationTimestamp(expirationTimestamp);
```

Comment:
Could you explain what a plan expiration timestamp is? (not in the code necessarily... but in a reply comment maybe?)